controllers/controller_test.go

```diff
@@ -0,0 +1,461 @@
+package controllers
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/gorilla/mux"
+	"github.com/gror/models"
+)
+
+type CreateDockerFailImplTest struct {
+}
+
+type CreateDockerSuccessImplTest struct {
+}
+
+type GetItemIDFailImplTest struct {
+}
+
+type GetItemReqSuccessImplTest struct {
+}
+
+type GetListSuccessImplTest struct {
+}
+
+type GetListFailImplTest struct {
+}
+
+type UpdateIDFullSuccessImpl struct {
+}
+
+type UpdateIDSuccessImpl struct {
+}
+
+type UpdateIDFailImpl struct {
+}
+
+var testCaseCreateFail = []struct {
+	Method       string
+	URL          string
+	Message      string
+	JSONBody     string
+	expectStatus int
+}{
+	{
+		Method:       "POST",
+		URL:          "/docker/config",
+		Message:      "Unprocessable Entity error",
+		JSONBody:     "{\"systemInfo\": {\"grorVersion\": \"1.0.0\",\"name\": \"cocooncam\" }",
+		expectStatus: 422,
+	},
+}
+var testCaseCreateSuccess = []struct {
+	Method       string
+	URL          string
+	Message      string
+	JSONBody     string
+	expectStatus int
+}{
+	{
+		Method:       "POST",
+		URL:          "/docker/config",
+		Message:      "{\"message\":\"Successfully created!\"}",
+		JSONBody:     "{\"systemInfo\": {\"grorVersion\": \"1.0.0\",\"name\": \"cocooncam\"} }",
+		expectStatus: 200,
+	},
+}
+
+var testCaseGetItemIDFail = []struct {
+	ID           string
+	Method       string
+	URL          string
+	Message      string
+	expectStatus int
+}{
+	{
+		ID:           "1100",
+		Method:       "GET",
+		URL:          "/docker/config/1100",
+		Message:      "Invalid Id bad request",
+		expectStatus: 400,
+	},
+}
+
+var testCaseGetItemFullSuccess = []struct {
+	ID           string
+	Method       string
+	URL          string
+	expectStatus int
+}{
+	{
+		ID:           "5b28b442a90362768113e47e",
+		Method:       "GET",
+		URL:          "/docker/config/5b28b442a90362768113e47e",
+		expectStatus: 200,
+	},
+}
+
+var testCaseUpdateIDFullSuccess = []struct {
+	ID           string
+	Method       string
+	Message      string
+	URL          string
+	JSONBody     string
+	expectStatus int
+}{
+	{
+		ID:           "5b28b442a90362768113e47e",
+		Method:       "PUT",
+		Message:      "{\"message\":\"Successfully updated!\"}",
+		URL:          "/docker/config/5b28b442a90362768113e47e",
+		JSONBody:     "{\"systemInfo\": {\"grorVersion\": \"1.1.0\",\"name\": \"cocooncam\"} }",
+		expectStatus: 200,
+	},
+}
+
+var testCaseUpdateIDSuccess = []struct {
```

Comment:
testcase status is 422 and your testcase name is success?

Reply:
This test case stays that the ID is correct but jsonbody is bad while unmarshling